src/app/admin/websiteDataTable/WebsiteDataTable.tsx

```diff
@@ -39,38 +39,31 @@ const WebsiteDataTable: React.FC = () => {
     const [errorMessage, setErrorMessage] = useState<string | null>(null);
     const dataGridRef = useGridApiRef();
 
-    useEffect(() => {
-        setIsLoading(true);
+    const fetchAndSetWebsiteData = useCallback(async () => {
         setErrorMessage(null);
-        fetchWebsiteData()
-            .then((response) => setRows(response))
-            .catch((error) => {
-                void logErrorReturnLogId("Error with fetch: website data", error);
-                setErrorMessage("Error fetching data, please reload");
-            })
-            .finally(() => setIsLoading(false));
+        const { data: websiteData, error: websiteDataError } = await fetchWebsiteData();
+        if (websiteDataError) {
+            setRows([]);
+            setErrorMessage(`${websiteDataError.type}. Log ID: ${websiteDataError.logId}`);
+        } else {
+            setRows(websiteData);
+        }
+        setIsLoading(false);
     }, []);
 
+    useEffect(() => {
+        setIsLoading(true);
+        void fetchAndSetWebsiteData();
```

Comment:
I think we should include `setIsLoading(true)` inside `fetchAndSetWebsiteData` rather than here, OR move `setIsLoading(false)` outside `fetchAndSetWebsiteData`. What do you think?

Reply:
I've put setIsLoading(true) inside fetchAndSetWebsiteData